packages/astro/src/content/utils.ts

```diff
@@ -207,9 +206,6 @@ export async function loadContentConfig({
 		return new NotFoundError('Failed to resolve content config.');
 	} finally {
 		await tempConfigServer.close();
-		// Reset NODE_ENV to initial value
-		// Vite's `createServer()` sets NODE_ENV to 'development'!
-		process.env.NODE_ENV = nodeEnv;
```

Comment:
This removal undoes https://github.com/withastro/astro/pull/5700#pullrequestreview-1232968289 since we're already using Vite 4. Thought I'll make this change since this PR is config related.